Ops/src/OpsWorkerGenetic.cpp

```diff
@@ -147,66 +143,91 @@ void WorkerGenetic::setMutationProbability(double value)
     mMutationProbability = value;
 }
 
+//! @brief Set elites paremter
+void WorkerGenetic::setNumberOfElites(size_t value)
+{
+    mNumElites = value;
+}
+
 
 //! @brief Auxiliary function for selecting parents for crossover
 void WorkerGenetic::selectParents() {
-    double bestValue = std::numeric_limits<double>::max();
-    for(size_t i=0; i<mNumPoints; ++i) {
-        double value = mObjectives[i]*opsRand();
-        if(value < bestValue) {
-            mParent1 = i;
-            bestValue = value;
+    std::vector<size_t> sortedIds = getIdsSortedFromWorstToBest();
+
+    // Select parent 1
+    std::vector<size_t> competitors;
+    competitors.push_back(opsRand()*(mNumPoints));
```

Comment:
Small (very) risk that: `opsRand()*(mNumPoints)` will be equal to mNumPoints, in which case the competitor (I assume its an index) will be out of bounds. If opsRand() returns "exactly" 1.

Reply:
Maybe make another help function, that takes an argument for the max allowed value.
But in that case consider the rounding.
In this case the rounding to size_t will always be by truncation. So its unfair since the max value will almost never be reached. (But in the current code that is a good thing to avoid out of bounds)

Reply:
I replaced the rand() function with C++11 generator, which generates a value in [0,1[ instead of [0,1]. This should eliminate the risk.